### PR TITLE
refactor light fixture placing

### PR DESCRIPTION
**Pull request by zjdtmkhzt**
## About the PR 
Refactors light fixture placement code a bit to move the stuff out of the walls and floors and into the light fixture object.
## Why's this needed? 
* The initial impetus for this was to make it possible to simply place light fixtures onto windows.
* Light fixtures should now also be offset correctly depending on the wall sprite, like those that are defined in maps already are.
* Removes a lot of sort of duplicate code?
* Ideally it will also prevent the thing where you smack walls with the fixtures while placing them.
## Changelog 

```changelog
(u) zjdtmkhzt
(+) Light fixtures can now be placed directly on windows.
```

### CODE REVIEW

**Comment by zjdtmkhzt** [comment]
I still have some minor merge conflicts to resolve and want to test it a bit before then, after that it should be good to merge hopefully.

Also, the RCD will still smack things with this, but I'd rather change that in a subsequent PR since this is already sizable and I might have to shuffle some RCD code around.

**Comment by ZeWaka** [comment]
finally, nice

File: code/modules/power/lighting.dm [hunk 1]
```diff
@@ -52,6 +55,70 @@ TYPEINFO(/obj/item/light_parts)
...
+			return FALSE
+		dir = get_dir(get_turf(target), user)
+		if (!is_cardinal(dir))
+			boutput(user, "You can't seem to reach that part of \the [target]. Try standing right up against it.")
+			return FALSE
+	if (locate(/obj/machinery/light) in get_turf(target))
+		boutput(user, "There's already a lamp there!")
+		return FALSE
+	return TRUE
+s
```
Comment (ZeWaka) [inline code/modules/power/lighting.dm:78]: s

Comment (zjdtmkhzt) [reply code/modules/power/lighting.dm:78]: s

Comment (ZeWaka) [reply code/modules/power/lighting.dm:78]: pls no s

**Comment by zjdtmkhzt** [comment]
Also added functionality now for stacking several lights on one tile, if it makes sense with the types and directions.
image

File: code/modules/power/lighting.dm (hunk 1 above)
Comment (zjdtmkhzt) [reply code/modules/power/lighting.dm]: ok

**Review by flappybatpal — APPROVED** [review]
Tested ingame and seems good.

**Pull request merged**